papermage/magelib/entity.py

```diff
@@ -46,10 +46,11 @@ def to_json(self) -> Dict:
 
     @classmethod
     def from_json(cls, entity_json: Dict) -> "Entity":
+        # the .get(..., None) or [] pattern is to handle the case where the key is present but the value is None
         return cls(
-            spans=[Span.from_json(span_json=span_json) for span_json in entity_json.get("spans", [])],
-            boxes=[Box.from_json(box_json=box_json) for box_json in entity_json.get("boxes", [])],
-            metadata=Metadata.from_json(entity_json.get("metadata", {})),
+            spans=[Span.from_json(span_json=span_json) for span_json in entity_json.get("spans", None) or []],
```

Comment:
can we avoid the `None` behavior? i think it's broken ppl's iterations in the past

Reply:
so it should just fail if metadata is set to none?